glib/src/log.rs

```diff
@@ -328,27 +336,131 @@ pub fn log_default_handler(log_domain: Option<&str>, log_level: LogLevel, messag
     }
 }
 
+// rustdoc-stripper-ignore-next
+/// Structure representing a single field in a structured log entry.
+///
+/// See [`g_log_structured`][gls] for details. Log fields may contain UTF-8 strings, binary with
+/// embedded nul bytes, or arbitrary pointers.
+///
+/// [gls]: https://docs.gtk.org/glib/func.log_structured.html
+#[cfg(any(feature = "v2_50", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_50")))]
+#[repr(transparent)]
+#[derive(Debug)]
+#[doc(alias = "GLogField")]
+pub struct LogField<'a>(ffi::GLogField, std::marker::PhantomData<&'a GStr>);
+
+#[cfg(any(feature = "v2_50", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_50")))]
+impl<'a> LogField<'a> {
+    // rustdoc-stripper-ignore-next
+    /// Creates a field from a borrowed key and value.
+    pub fn new(key: &'a GStr, value: &[u8]) -> Self {
+        Self(
+            ffi::GLogField {
+                key: key.as_ptr(),
+                value: value.as_ptr() as *const _,
+                length: value.len().try_into().unwrap(),
+            },
+            Default::default(),
+        )
+    }
+    // rustdoc-stripper-ignore-next
+    /// Creates a field from a raw pointer. Can be used to pass custom data into a writer function
+    /// set with [`log_set_writer_func`], where it can be retreived using [`value_ptr`]. Panics if
+    /// `key` contains null bytes.
+    pub fn new_ptr<T>(key: &'a GStr, ptr: *const T) -> Self {
+        Self(
+            ffi::GLogField {
+                key: key.as_ptr(),
+                value: ptr as *const _,
+                length: 0,
+            },
+            Default::default(),
+        )
+    }
+    // rustdoc-stripper-ignore-next
+    /// Retreives the field key.
+    pub fn key(&self) -> &str {
+        unsafe { std::ffi::CStr::from_ptr(self.0.key as *const i8) }
+            .to_str()
+            .unwrap()
+    }
+    // rustdoc-stripper-ignore-next
+    /// Retrieves the raw pointer of the field value. This should only be used on fields created
+    /// with [`new_ptr`] inside writer functions set with [`log_set_writer_func`].
+    pub fn value_ptr<T>(&self) -> *const T {
```

Comment:
Same here: we could return `Option<*const T>`, returning `None` of `lenght != 0`

Reply:
I'll do this and also change it to a `usize`. If someone wants to unsafely transmute it back to a pointer that will still work